packages/docgen/src/get-jsdoc-from-statement.js

```diff
@@ -0,0 +1,363 @@
+/**
+ * External dependencies.
+ */
+const ts = require( 'typescript' );
+const { SyntaxKind } = ts;
+
+const typeToString = require( './type-to-string' );
+
+/**
+ * @typedef {import('typescript').Statement} Statement
+ */
+
+// from https://stackoverflow.com/questions/9716468/pure-javascript-a-function-like-jquerys-isnumeric
+function isNumeric( n ) {
+	return ! isNaN( parseFloat( n ) ) && isFinite( n );
+}
+
+function stripQuotes( str ) {
+	const first = str[ 0 ];
+	const last = str[ str.length - 1 ];
+
+	if (
+		( first === `'` && last === `'` ) ||
+		( first === '"' && last === '"' ) ||
+		( first === '`' && last === '`' )
+	) {
+		return str.substring( 1, str.length - 1 );
+	}
+
+	return str;
+}
+
+const getType = ( typeExpression ) =>
+	typeExpression ? typeToString( typeExpression.type ) : 'undocumented';
+
+/**
+ * It loops from = character until ] is found and extract default value from it.
+ * If it starts with a quote(', ", `), it tries to find it.
+ * This function assumes that the name of param tag always ends with ].
+ *
+ * @param {string} code
+ * @param {number} nameEndPos
+ */
+const getDefaultValue = ( code, nameEndPos ) => {
+	let closingBracketPos = nameEndPos;
+
+	let startPos = nameEndPos + 1;
+	let endingChar = ']';
+	const firstChar = code[ startPos ];
+
+	if ( firstChar === `'` || firstChar === '"' || firstChar === '`' ) {
+		endingChar = firstChar;
+		startPos = nameEndPos + 2;
+	}
+
+	if ( firstChar !== '[' ) {
+		for ( let i = startPos; ; i++ ) {
+			if ( code[ i ] === '\\' && code[ i + 1 ] !== ']' ) {
+				continue;
+			}
+
+			if (
+				code[ i ] === endingChar ||
+				code[ i ] === '\n' ||
+				code[ i ] === '\r'
+			) {
+				closingBracketPos = i;
+				break;
+			}
+		}
+	} else {
+		// count the broken pairs of brackets
+		let broken = 0;
+
+		for ( let i = startPos; ; i++ ) {
+			if (
+				( code[ i ] === ']' && broken === 0 ) ||
+				code[ i ] === '\n' ||
+				code[ i ] === '\r'
+			) {
+				closingBracketPos = i;
+				break;
+			}
+
+			if ( code[ i ] === '[' ) {
+				broken++;
+			}
+
+			if ( code[ i ] === ']' ) {
+				broken--;
+			}
+		}
+	}
+
+	const rawValue = code.substring( startPos, closingBracketPos );
+
+	if ( isNumeric( rawValue ) ) {
+		if ( Number.isInteger( rawValue ) ) {
+			return parseInt( rawValue );
+		}
+
+		return parseFloat( rawValue );
+	}
+
+	if ( rawValue === 'null' ) return null;
+	if ( rawValue === 'true' ) return true;
+	if ( rawValue === 'false' ) return false;
+
+	if ( rawValue[ 0 ] === '{' && rawValue[ rawValue.length - 1 ] === '}' ) {
+	}
```

Comment:
Empty conditional block here?

Reply:
I first thought about parsing objects and arrays. But I realized that to add them to docs, I need to return them back to strings. And it's unnecessary. So, I gave it up. 

But I forgot to remove the code for that attempt.